add cooc gcp to xsoar and xsiam (#44449)
demisto/content: Packs/GCP/ReleaseNotes/1_2_0.md

#### Integrations

##### Google Cloud Platform

> - Added support for **Cortex XSOAR** and **Cortex XSIAM** (version < 3.0) marketplace deployments. The integration can now be configured with a GCP Service Account private key JSON and a default *GCP Project ID* for use outside of Cortex Cloud or Cortex XSIAM (version >= 3.0).

Reviewer: ```suggestion
- Added support for **Cortex XSOAR** and **Cortex XSIAM** (version < 3.0) Marketplace deployments. The integration can now be configured with a GCP Service Account private key JSON and a default *GCP Project ID*.
```

> - Updated the *Service Account Private Key (JSON)* parameter description to clarify that it is required for Cortex XSOAR and Cortex XSIAM, and optional on Cortex Cloud or Cortex XSIAM (version >= 3.0).

Reviewer: ```suggestion
- Updated the *Service Account Private Key (JSON)* parameter description to clarify that it is required for Cortex XSOAR and Cortex XSIAM (version < 3.0), and optional for Cortex Cloud or Cortex XSIAM (version >= 3.0).
```

> - Added the *GCP Project ID* integration parameter, used by the test button on Cortex XSOAR and Cortex XSIAM to validate credentials.

Reviewer: ```suggestion
- Added the *GCP Project ID* integration parameter, used by the **Test** button for Cortex XSOAR and Cortex XSIAM (version < 3.0) to validate credentials.
```

> - Updated the ***test-module*** command to call the Cloud Resource Manager `testIamPermissions` API directly when running on Cortex XSOAR or Cortex XSIAM, validating that the provided service account credentials are functional.

Reviewer: ```suggestion
- Updated the ***test-module*** command to call the Cloud Resource Manager `testIamPermissions` API directly when running on Cortex XSOAR or Cortex XSIAM (version < 3.0), validating the provided service account credentials are functional.
```

> - Updated the integration description to cover all three supported platforms.

Reviewer: ```suggestion
- Updated the integration description to cover the supported platforms.
```

> - Updated the *project_id* arguments in all commands to `required: false` with `required:platform: true`, making the argument optional on Cortex XSOAR and Cortex XSIAM (where it can be inferred from the service account JSON) while remaining required on Cortex Cloud (platform) or Cortex XSIAM (version >= 3.0).

Reviewer: ```suggestion
- Updated the *project_id* arguments in all commands to make the argument optional on Cortex XSOAR and Cortex XSIAM (version < 3.0), while remaining required on Cortex Cloud or Cortex XSIAM (version >= 3.0).
```
